### ISSUE TITLE
type generation on enum fails

### ISSUE THREAD

**Issue by ChristopherDavenport** [guardrail-dev/guardrail#1029]
Schema declaration will fail as the EnumContainer will reference MyEnumType, but no type exists. Either a type alias in the package object or some type of coproduct is necessary to get the code to compile. Behavior same on client and server for both http4s and akka-http

```yaml
# This is in the components/schemas section
MyEnumType:
  type: integer
  enum:
    - 0
    - 1
    - 2
    - 3
 EnumContainer:
   type: object
   properties:
     myEnum:
       $ref: "#/components/schemas/MyEnumType"
 ```
 
 Generates
 
 ```scala
 case class EnumContainer(myEnum: Option[MyEnumType] = None)
 ```
 
 But no `MyEnumType` was generated.
 
 Can be fixed with 
```scala
 package object definitions {
   type MyEnumType = Int
   ... rest of code-gen already here
}
```

**Comment by blast-hardcheese** [comment]
Taking a look.

Background on this particular problem: swagger didn't support non-string enumerations for the longest time, the spec additionally currently doesn't support labeled mappings to the best of my knowledge. We have an extension, `x-jvm-type: com.example.MyEnum`, which can be used to overload the generated types if they are insufficient. In your case,
```yaml
properties:
  myEnum:
    type: integer
    x-jvm-type: com.example.MyEnum
```
will generate

```scala
case class EnumContainer(myEnum: Option[com.example.MyEnum] = None)
```

**Comment by ChristopherDavenport** [comment]
I could definitely link to another class. I'm just thinking that at a minimum a type alias would allow code to compile on Int based enumerations. Trying to reuse existing swagger with limited changes that are guardrail specific at least on this first pass.

**Comment by blast-hardcheese** [comment]
Certainly. I would have expected what you have to work similarly to this test: https://github.com/guardrail-dev/guardrail/blob/1d50fa3d957708c0aa72d811dd8725ea47f0d117/modules/codegen/src/test/scala/tests/generators/akkaHttp/EnumTest.scala#L48-L54

**Comment by blast-hardcheese** [comment]
my thinking is that the `type: integer` is throwing this off, but I'd also have expected generation to fail

**Comment by blast-hardcheese** [comment]
More tests required, but I believe that will resolve the issue.

**Comment by blast-hardcheese** [comment]
This has been resolved in 0.63.0, thanks for your patience!
